[OSDOCS#18857]: Added a new module for Fencing Validator script (#110148)
openshift/openshift-docs: modules/Installation-fencing-validator-generating-help.adoc

:_mod-docs-content-type: REFERENCE
[id="installation-fencing-validator-cli-options_{context}"]
= Command-line options for fencing validator script

[role="_abstract"]

> You can see different command-line options for the fencing validator script by running the following command: 

Reviewer: Same question about the lead-in sentence for the abstract.

> [source,bash]

Reviewer: [source,terminal
----
$ oc debug node/\<node_name\> -- chroot /host /usr/local/bin/fencing\_validator --help
----

oc debug node/\<node_name\> -- chroot /host /usr/local/bin/fencing\_validator --help

> The command-line options are as follows:

Reviewer: ```suggestion
The following table details command-line options for the `oc debug node` command:
```

[cols="1,1,2", options="header"]
|===
| Flag | Argument | Description

| `--user`
| `<ssh_user>`
| SSH username for remote node access (default: `core`)

| `--ssh-key`
| `<path>`
| Path to SSH private key

| `--kubeconfig`
| `<path>`
| Path to kubeconfig file

| `--transport`
| `auto`, `ssh`, or `ocdebug`

> | How the script connects to the other node. For more information, see `auto` tries SSH first, falls back to `oc debug`

Reviewer: ```suggestion
| How the script connects to the other node. For more information, see `auto` tries SSH first, falls back to `oc debug`.
```

| `--timeout`
| `<seconds>`
| Maximum wait time for recovery loops. By default, it is 1200 seconds or 20 minutes.

| `--hosts`
| `"nodeA,nodeB"`

> | Comma-separated pair of node hostnames or IPs.

Reviewer: ```suggestion
| Comma-separated pair of node hostnames or IP addresses.
```

| `--host-a`
| `<host>`
| Explicitly set the first node.

| `--host-b`
| `<host>`
| Explicitly set the second node.

| `--disruptive`
| *(none)*
| Enable destructive fencing tests (reboots nodes)

| `--dry-run`
| *(none)*
| Show what would be done without doing actual validations.

| `-h`, `--help`
| *(none)*
| Show usage information.
|===

You can set all the flags by using the following environment variables:

* `SSH_USER`
* `SSH_KEY`
* `KUBECONFIG`
* `TRANSPORT`
* `TIMEOUT`
* `DISRUPTIVE`
* `DRY_RUN`

> * `IP_A / IP_B` — set host addresses directly

Reviewer: ```suggestion
* `IP_A / IP_B`: set host addresses directly
```

Do for other listed items.

> * `OC_BIN` — custom oc binary path

Reviewer: ```suggestion
* `OC_BIN` — custom `oc` binary path
```

> * `OC_REQ_TIMEOUT` — per-API-call timeout. By default, it is 10 seconds.

Reviewer: ```suggestion
* `OC_REQ_TIMEOUT` — per-API-call timeout. The default value is `10` seconds.
```

> * `CMD_EXEC_TIMEOUT_SECS` — per-command timeout. By default, it is 60 seconds.

Reviewer: ```suggestion
* `CMD_EXEC_TIMEOUT_SECS` — per-command timeout. The default value is `60` seconds.
```
